tests/run/i1169b.scala

```diff
@@ -0,0 +1,18 @@
+
+object Test {
+  def main(args: Array[String]): Unit = {
+    val xProducer = new XProducer
+    println(xProducer.produce.underlying)
```

Comment:
This code should not compile as our current Value-class encoding can not handle it. Details can be found here: https://github.com/lampepfl/dotty/issues/1169#issuecomment-196784201